modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/display/context/CommerceDashboardForecastDisplayContext.java

```diff
@@ -26,24 +34,61 @@ public class CommerceDashboardForecastDisplayContext {
 	public CommerceDashboardForecastDisplayContext(
 			ModelResourcePermission<AccountEntry>
 				accountEntryModelResourcePermission,
+			AssetCategoryLocalService assetCategoryLocalService,
+			ConfigurationProvider configurationProvider,
 			HttpServletRequest httpServletRequest)
 		throws PortalException {
 
 		_accountEntryModelResourcePermission =
 			accountEntryModelResourcePermission;
+		_assetCategoryLocalService = assetCategoryLocalService;
 
 		_commerceDashboardForecastRequestHelper =
 			new CommerceDashboardForecastRequestHelper(httpServletRequest);
 
 		_commerceDashboardForecastPortletInstanceConfiguration =
-			ConfigurationProviderUtil.getPortletInstanceConfiguration(
+			configurationProvider.getPortletInstanceConfiguration(
 				CommerceDashboardForecastPortletInstanceConfiguration.class,
 				_commerceDashboardForecastRequestHelper.getThemeDisplay());
 	}
 
 	public String getAssetCategoryIds() {
-		return _commerceDashboardForecastPortletInstanceConfiguration.
-			assetCategoryIds();
+		String[] assetCategoryExternalReferenceCodes =
+			_commerceDashboardForecastPortletInstanceConfiguration.
+				assetCategoryExternalReferenceCodes(
+				).split(
+					StringPool.COMMA
+				);
+
+		List<String> assetCategoryIds = new ArrayList<>();
+
+		ThemeDisplay themeDisplay =
+			_commerceDashboardForecastRequestHelper.getThemeDisplay();
+
+		for (String assetCategoryExternalReferenceCode :
+				assetCategoryExternalReferenceCodes) {
+
+			AssetCategory assetCategory =
+				_assetCategoryLocalService.
+					fetchAssetCategoryByExternalReferenceCode(
+						assetCategoryExternalReferenceCode,
+						themeDisplay.getScopeGroupId());
+
+			if (assetCategory == null) {
+				assetCategory =
+					_assetCategoryLocalService.
+						fetchAssetCategoryByExternalReferenceCode(
+							assetCategoryExternalReferenceCode,
+							themeDisplay.getCompanyGroupId());
+			}
+
+			if (assetCategory != null) {
+				assetCategoryIds.add(
+					String.valueOf(assetCategory.getCategoryId()));
+			}
+		}
+
+		return StringUtil.merge(assetCategoryIds);
```

Comment:
Burrito

cc @smottal

Reply:
I stopped to check immediately for this chain
```
_commerceDashboardForecastPortletInstanceConfiguration.
				assetCategoryExternalReferenceCodes(
				).split(
					StringPool.COMMA
				)
```

Reply:
and I think you can do it using TransformUtil